source/Sashimi.AzureWebApp/Endpoints/AzureWebAppDeploymentTargetTypeProvider.cs

```diff
@@ -44,6 +44,23 @@ public IEnumerable<(string key, object value)> GetFeatureUsage(IEndpointMetricCo
             yield return ("azurewebapps", total);
         }
 
+        public IEnumerable<ScriptFunctionRegistration> GetScriptFunctionRegistrations()
+        {
+            yield return new ScriptFunctionRegistration("OctopusAzureWebAppTarget",
```

Comment:
I really like the general direction this is heading. I wonder if there'd be benefit in having this method in the service message handler though? That handler is the one that has to interpret what this method creates, so it might be better in the same class. For example, the parameter names might share a string literal once they're in the same class?